tests/src/rules/no-cycle.js

```diff
@@ -32,7 +32,7 @@ ruleTester.run('no-cycle', rule, {
     test({ code: 'var foo = require("./")' }),
     test({ code: 'var foo = require("@scope/foo")' }),
     test({ code: 'var bar = require("./bar/index")' }),
-    test({ code: 'var bar = require("./bar")' }),
+    ...usingFlatConfig ? [] : [test({ code: 'var bar = require("./bar")' })],
```

Comment:
why can't this one work in flat config?

Reply:
As it's considered a duplicate of the next line and v9 ruletester doesn't allow duplicate tests

Reply:
it's not, though - it has a different filename. v9 doesn't consider that to be different? sounds like a bug in v9.

Reply:
Yup, feel free to raise an issue over at eslint

Reply:
in that case, let's add a code comment or something so we can keep both test cases.

Reply:
ah ha no this does not actually have a different filename!

This test defines its own `test` function which always overrides the filename:

```
const test = (def) => _test(Object.assign(def, {
  filename: testFilePath('./cycles/depth-zero.js'),
}));
```

I've pushed up 5de28453 changing this which should be fine to cherry-pick